witness/golang/omniwitness/usbarmory/internal/storage/slots/slots.go

```diff
@@ -117,14 +117,11 @@ func (p *Partition) Open(slot uint) (*Slot, error) {
 		return nil, fmt.Errorf("invalid slot %d (partition has %d slots)", slot, l)
 	}
 	s := &p.slots[slot]
-	if s.journal == nil {
-		glog.V(2).Infof("Opening journal for slot %d", slot)
-		j, err := OpenJournal(p.dev, s.start, s.length, p.sha256)
-		if err != nil {
-			return nil, fmt.Errorf("failed to open journal for slot %d: %v", slot, err)
-		}
-		s.journal = j
+	glog.V(2).Infof("Opening slot %d", slot)
+	if err := s.Open(p.dev, p.sha256); err != nil {
+		glog.V(2).Infof("Failed to open slot %d: %v", slot, err)
```

Comment:
This should probably return the error too?

Reply:
The method comment does cover that case already :)